src/qibolab/platforms/platform.py

```diff
@@ -32,16 +36,29 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None, raw_adc=False):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
-        return self.design.play(self.qubits, sequence, nshots=nshots, relaxation_time=relaxation_time, raw_adc=raw_adc)
+    def execute_pulse_sequence(self, sequence, **kwargs):
+        options = ExecutionParameters(**kwargs)
```

Comment:
Consider asking directly `ExecutionParameters` in the input. It is better to have a clear typing and limit internal conversions, whenever possible.
```suggestion
    def execute_pulse_sequence(self, sequence, options: ExecutionParameters):
```

Reply:
If I ask it in the input that means changing how qibocal routines look from
```
# execute the pulse sequence
        results = platform.execute_pulse_sequence(
            sequence,
            nshots=params.nshots,
            relaxation_time=params.relaxation_time,
            acquisition_type=AcquisitionType.INTEGRATION,
            averaging_mode=AveragingMode.CYCLIC,
        )
```
to something like this
```
options = {
    "nshots": params.nshots,
    "relaxation_time": params.relaxation_time,
    "acquisition_type": AcquisitionType.INTEGRATION,
    "averaging_mode": AveragingMode.CYCLIC, 
}

ExecutionParameters = (options)

results = platform.execute_pulse_sequence(
    sequence,
    options,
)
```
And I'm not sure if changing routines like this would be to everyone's liking

Reply:
We will put it on qibocal @aorgazf @stavros11 if I understood correctly from the meeting we just had

Reply:
You can even have a closer shape, instead of:
```py
# execute the pulse sequence
        results = platform.execute_pulse_sequence(
            sequence,
            nshots=params.nshots,
            relaxation_time=params.relaxation_time,
            acquisition_type=AcquisitionType.INTEGRATION,
            averaging_mode=AveragingMode.CYCLIC,
        )
```

you can do:
```py
# execute the pulse sequence
        results = platform.execute_pulse_sequence(
            sequence,
            ExecutionParameters(
                nshots=params.nshots,
                relaxation_time=params.relaxation_time,
                acquisition_type=AcquisitionType.INTEGRATION,
                averaging_mode=AveragingMode.CYCLIC,
            )
        )
```

The difference is not incredible, but feel free to ask if people prefer keeping the old one. At the moment, we're not fully consistent with types anyhow, so there is not an urge to use them everywhere.

Reply:
Fine by me :)